src/StarRating/StarRatingControl.ts

```diff
@@ -0,0 +1,240 @@
+﻿module Fayde.Controls {
+    import Control = Fayde.Controls.Control;
+    import GridUnitType = minerva.controls.grid.GridUnitType;
+
+    // Defines a datepicker in the format of windows phone pickers (Touch friendly)
+    export class StarRatingControl extends Control {
+        private stars = new Array<Star>();
+        private isHovering: boolean = false;
+        private LayoutRootStarList: Fayde.Controls.Grid;
+
+        //NumberOfStarsProperty
+        static NumberOfStarsProperty = DependencyProperty.Register("NumberOfStars",() => Number, StarRatingControl, 5,(d, args) => (<StarRatingControl>d).OnNumberOfStarsChanged(args));
+        NumberOfStars: number;
+
+        OnNumberOfStarsChanged(e: DependencyPropertyChangedEventArgs) {
+            this.CreateStars();
+            this.RefreshStarRating();
+        }       
+
+        //RatingProperty
+        static RatingProperty = DependencyProperty.Register("Rating",() => Number, StarRatingControl, 0,(d, args) => (<StarRatingControl>d).OnRatingChanged(args));
+        Rating: number;
+
+        OnRatingChanged(e: DependencyPropertyChangedEventArgs) {
+            this.DrawUnhovered();
+        }
+
+        //HoverRatingProperty
+        static HoverRatingProperty = DependencyProperty.Register("HoverRating",() => Number, StarRatingControl, 0,(d, args) => (<StarRatingControl>d).OnHoverRatingChanged(args));
+        HoverRating: number;
+
+        OnHoverRatingChanged(e: DependencyPropertyChangedEventArgs) {
+            this.RefreshStarRating();
+        }
+
+        //StarFillBrushProperty
+        static StarFillBrushProperty = DependencyProperty.Register("StarFillBrush",() => Fayde.Media.Brush, StarRatingControl,(new Fayde.Media.SolidColorBrush(Color.FromRgba(0xFF, 0xFF, 0, 0xFF))),(d, args) => (<StarRatingControl>d).StarFillBrushChanged(args));
```

Comment:
Setting default brush values has bad side effects.  If someone tries to adjust the `Color` property of this Brush it will adjust for every StarRating.
Also, these defaults are best set in the theme file as Setters.